packages/neon_framework/packages/notifications_push_repository/lib/src/notifications_push_repository.dart

```diff
@@ -0,0 +1,243 @@
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:account_repository/account_repository.dart';
+import 'package:built_collection/built_collection.dart';
+import 'package:collection/collection.dart';
+import 'package:crypton/crypton.dart';
+import 'package:logging/logging.dart';
+import 'package:nextcloud/nextcloud.dart';
+import 'package:nextcloud/notifications.dart' as notifications;
+import 'package:notifications_push_repository/src/models/models.dart';
+import 'package:notifications_push_repository/src/notifications_push_storage.dart';
+import 'package:notifications_push_repository/src/utils/encryption.dart';
+import 'package:unifiedpush/unifiedpush.dart';
+
+/// Signature of the callback triggered by UnifiedPush when a new push notification is received.
+typedef OnMessageCallback = void Function(Uint8List message, String accountID);
+
+final _log = Logger('NotificationsPushRepository');
+
+/// A repository for managing push notifications and subscriptions.
+final class NotificationsPushRepository {
+  /// Creates a new [NotificationsPushRepository].
+  NotificationsPushRepository({
+    required AccountRepository accountRepository,
+    required NotificationsPushStorage storage,
+    required OnMessageCallback onMessage,
+  })  : _accountRepository = accountRepository,
+        _storage = storage,
+        _onMessage = onMessage;
+
+  final AccountRepository _accountRepository;
+  final NotificationsPushStorage _storage;
+  final OnMessageCallback _onMessage;
```

Comment:
I'm not sure about this callback.
Why does this return an unencrypted message? This way, the caller needs to also handle the storage for decryption.

Reply:
You mean encrypted message?
I really wanted to change this, but the problem is that for this callback to work it needs to be a top-level or static method.
This makes it really painful as we can not modify the data (e.g. decrypt and parse) before passing it on.
I don't see how this can be done in a better way than it is right now :/

Reply:
Meh, I thought that might be the case.
Let's keep it for now, although I really dislike having to do this :(